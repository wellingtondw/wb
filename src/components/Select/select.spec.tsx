import { renderWithTheme } from "../../utils/tests/helper";

import { Select } from ".";
import { fireEvent, screen } from "@testing-library/react";

const options = [
  { label: "100km", value: "100" },
  { label: "200km", value: "200" },
  { label: "300km", value: "300" },
];

describe("<Select />", () => {
  it("should be able to render correctly", () => {
    const { container } = renderWithTheme(
      <Select label="label" options={options} onChange={jest.fn()} />
    );

    expect(screen.getByText("label")).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should be able to render disabled", () => {
    renderWithTheme(
      <Select label="label" options={options} onChange={jest.fn()} disabled />
    );
    const selectElement = screen.getByRole("button");

    expect(selectElement).toHaveStyle({
      opacity: 0.6,
      cursor: "not-allowed",
    });

    fireEvent.click(selectElement);

    expect(screen.getByTestId("drop").getAttribute("aria-hidden")).toBe("true");
  });

  it("should be able to render and select options", () => {
    const onChange = jest.fn();
    renderWithTheme(
      <Select label="label" options={options} onChange={onChange} />
    );

    const selectElement = screen.getByRole("button");
    expect(screen.getAllByRole("button")).toHaveLength(1);

    fireEvent.click(selectElement);

    expect(screen.getByTestId("drop").getAttribute("aria-hidden")).toBe(
      "false"
    );
    expect(screen.getAllByRole("button")).toHaveLength(4);

    fireEvent.click(screen.getByRole("button", { name: /100km/i }));

    expect(screen.getByTestId("drop").getAttribute("aria-hidden")).toBe("true");
    expect(selectElement).toHaveTextContent("label: 100km");
    expect(onChange).toHaveBeenCalledWith("100");
  });
});
