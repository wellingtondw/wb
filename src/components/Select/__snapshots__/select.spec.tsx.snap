// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should be able to render correctly 1`] = `
<div
  class="sc-bczRLJ bzCupB"
  role="select"
>
  <button
    class="sc-gsnTZi hzeSGz"
  >
    <label
      class="sc-dkzDqf fwemrT"
    >
      label
    </label>
  </button>
  <div
    aria-hidden="true"
    class="sc-hKMtZM jAyrcp"
    data-testid="drop"
  >
    <button
      class="sc-eCYdqJ gSutqM"
    >
      100km
    </button>
    <button
      class="sc-eCYdqJ gSutqM"
    >
      200km
    </button>
    <button
      class="sc-eCYdqJ gSutqM"
    >
      300km
    </button>
  </div>
</div>
`;
